core/src/main/java/com/sequenceiq/cloudbreak/service/template/ClusterTemplateService.java

```diff
@@ -313,16 +337,17 @@ private void updateDefaultClusterTemplates(Workspace workspace) {
             LOGGER.debug("Outdated clusterTemplates deleted: '{}'.", outdatedTemplates.size());
             clusterTemplates = clusterTemplateRepository.findAllByNotDeletedInWorkspace(workspace.getId());
             LOGGER.debug("None deleted clusterTemplates collected: '{}'.", clusterTemplates.size());
-            clusterTemplateLoaderService.loadClusterTemplatesForWorkspace(clusterTemplates, this::createAll);
+            clusterTemplateLoaderService.loadClusterTemplatesForWorkspace(clusterTemplates, workspace, this::createAll);
             LOGGER.debug("ClusterTemplate modifications finished based on the defaults for '{}' workspace.", workspace.getId());
         }
     }
 
     private Collection<ClusterTemplate> createAll(Iterable<ClusterTemplate> clusterTemplates) {
-        return StreamSupport.stream(clusterTemplates.spliterator(), false)
+        User user = userService.getOrCreate(legacyRestRequestThreadLocalService.getCloudbreakUser());
+        return StreamSupport.stream(clusterTemplates.spliterator(), true)
```

Comment:
Can we set it to false as it was? As I know it is not a very good idea to use parallal streams in a multithreaded web based application because it can cause high load for one request and the application should be able to serve many request at the same time.